lib/prime.rb

```diff
@@ -30,17 +30,44 @@ def prime_division(generator = Prime::Generator23.new)
     Prime.prime_division(self, generator)
   end
 
+
+
   # Returns true if +self+ is a prime number, else returns false.
   def prime?
-    return self >= 2 if self <= 3
-    return true if self == 5
-    return false unless 30.gcd(self) == 1
-    (7..Integer.sqrt(self)).step(30) do |p|
-      return false if
-        self%(p)    == 0 || self%(p+4)  == 0 || self%(p+6)  == 0 || self%(p+10) == 0 ||
-        self%(p+12) == 0 || self%(p+16) == 0 || self%(p+22) == 0 || self%(p+24) == 0
-    end
-    true
+    # https://en.wikipedia.org/wiki/Miller%E2%80%93Rabin_primality_test
+    # https://arxiv.org/pdf/1509.00864.pdf
+    #
+    # if n < 3,317,044,064,679,887,385,961,981,
+    # it is enough to test a = 2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, and 41.
+    # and the result will be deterministic.
+
+    return false if self < 2
+    return true  if self < 4
```

Comment:
You can merge the initial tests using ``return self > 1 if self < 4``.
This will do one extra comparison for n < 2, but only one for n > 4.
I guess most use cases only deal with big positive numbers.
Not sure if this makes any difference, just noticed it :smile:.

Reply:
Merged.

Reply:
Merged.